src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -563,6 +562,7 @@ void DeviceControllerSystemState::Shutdown()
 
     // Shut down the interaction model
     app::InteractionModelEngine::GetInstance()->Shutdown();
+    app::InteractionModelEngine::GetInstance()->SetDataModelProvider(nullptr);
```

Comment:
@tcarmelveilleux Could you double-check that your approval still stands after these changes?

Reply:
ACK, OK with change.